drivers/audio/dmic_nrfx_pdm.c

```diff
@@ -61,6 +70,35 @@ static void stop_pdm(struct dmic_nrfx_pdm_drv_data *drv_data)
 	nrfx_pdm_stop(drv_data->pdm);
 }
 
+static int request_clock(struct dmic_nrfx_pdm_drv_data *drv_data)
+{
+	if (!drv_data->request_clock) {
+		return 0;
+	}
+#if CONFIG_CLOCK_CONTROL_NRF
+	return onoff_request(drv_data->clk_mgr, &drv_data->clk_cli);
+#elif CONFIG_CLOCK_CONTROL_NRF2_AUDIOPLL
+	return nrf_clock_control_request(drv_data->audiopll_dev, NULL, &drv_data->clk_cli);
+#else
+	return 0;
```

Comment:
Should't be `-ENOTSUP` ? same [here](https://github.com/nrfconnect/sdk-zephyr/pull/2623/files#diff-794b53baceb847386ff20b54cc073b4df67825695c91353cc0226844d4467743R98)